simulator/gdb/gdb_interface.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * gdb_interface.c - Functional simulator interfaces for GDB
+ * @author Vyacheslav Kompan
+ * Copyright 2018 MIPT-MIPS
+ */
+
+
+/* BFD config */
+#include <config.h>
+
+/* Simulator Interface */
+#include "sim_wrapper.h"
+
+/* GDB Interface */
+#include <gdb/remote-sim.h>
+#include <gdb/callback.h>
+/* GDB Simulator utilities */
+#include <sim-config.h>
+#include <sim-types.h>
+#include <sim-inline.h>
+#include <sim-arange.h>
+#include <sim-base.h>
+
+
+SIM_DESC sim_open (SIM_OPEN_KIND kind, struct host_callback_struct *callback,
+                   struct bfd *abfd, char *const *argv) {
```

Comment:
Could you please add a description of `abfd`? Other names looks clear.

Reply:
This functions are described [here](https://github.com/MIPT-ILab/mipt-sim-gdb/blob/gdb-8.1-branch/include/gdb/remote-sim.h). Should I copy the description for it?

Reply:
~~Yes.~~
Well, that source is distributed by GNU GPL 3 which is incompatible with MIT (you may include MIT sources to GNU GPL 3, but not vice versa). So it's better to explain it in your own words.